src/module.c

```diff
@@ -2717,7 +2722,10 @@ RedisModuleString *RM_CreateStringFromStreamID(RedisModuleCtx *ctx, const RedisM
  * pass ctx as NULL when releasing the string (but passing a context will not
  * create any issue). Strings created with a context should be freed also passing
  * the context, so if you want to free a string out of context later, make sure
- * to create it using a NULL context. */
+ * to create it using a NULL context.
+ * 
+ * This API is not thread safe, access to these retained strings must
+ * be done with GIL locked. */
```

Comment:
```suggestion
 * This API is not thread safe, access to these retained strings (if they originated
 *  from a client command arguments) must be done with GIL locked. */
```

if you agree, mirror to the other APIs with that comment.